Java/PoolAndResourceFile/src/main/java/PoolAndResourceFile.java

```diff
@@ -173,27 +193,53 @@ private static String uploadFileToCloud(CloudBlobContainer container, String fil
         blob.upload(new FileInputStream(source), source.length());
 
         // Create policy with 1 day read permission
-        SharedAccessBlobPolicy policy = new SharedAccessBlobPolicy();
-        EnumSet<SharedAccessBlobPermissions> perEnumSet = EnumSet.of(SharedAccessBlobPermissions.READ);
-        policy.setPermissions(perEnumSet);
-
-        Calendar c = Calendar.getInstance();
-        c.setTime(new Date());
-        c.add(Calendar.DATE, 1);
-        policy.setSharedAccessExpiryTime(c.getTime());
+        SharedAccessBlobPolicy policy = getSharedAccessBlobPolicy(SharedAccessBlobPermissions.READ, 1);
 
         // Create SAS key
         String sas = blob.generateSharedAccessSignature(policy, null);
         return blob.getUri() + "?" + sas;
     }
 
+    /**
+     *
+     * @param container
+     *          container to upload the file
+     * @param filePattern
+     *          all the files matching the given pattern will be upload to container
+     * @return  list of output files
+     * @throws StorageException
+     * @throws InvalidKeyException
+     */
+    private static List<OutputFile> getOutputFiles(CloudBlobContainer container, String filePattern) throws StorageException, InvalidKeyException {
+        // Create policy with 1 day write permission
+        SharedAccessBlobPolicy policy = getSharedAccessBlobPolicy(SharedAccessBlobPermissions.WRITE, 1);
+        String sasToken = container.generateSharedAccessSignature(policy, null);
+        String containerSasUrl = String.format("%s?%s", container.getUri().toString(), sasToken);
+        OutputFileBlobContainerDestination outputFileBlobContainerDestination =
+                new OutputFileBlobContainerDestination()
+                        .withContainerUrl(containerSasUrl);
+
+        List<OutputFile> outputFiles = new ArrayList<>();
+        OutputFile outputFile = new OutputFile()
+                .withFilePattern(filePattern)
+                .withDestination(new OutputFileDestination()
+                        .withContainer(outputFileBlobContainerDestination))
+                .withUploadOptions(new OutputFileUploadOptions()
+                        .withUploadCondition(OutputFileUploadCondition.TASK_COMPLETION));
+
+        outputFiles.add(outputFile);
+        return outputFiles;
+    }
+
     /**
      * Create a job with a single task
      * 
      * @param client
      *            batch client instance
      * @param container
```

Comment:
rename this variable now that there are two containers

Reply:
renamed container to resourceFileUploadContainer